src/components/App.js

```diff
@@ -72,65 +130,103 @@ class App extends PureComponent {
           channels: [
             ...this.state.channels.slice(0, channelIndex),
             { ...this.state.channels[channelIndex], ...data },
-            ...this.state.channels.slice(channelIndex + 1),
+            ...this.state.channels.slice(channelIndex + 1)
           ]
         });
       }
     });
-  }
+  };
 
-  // Join selected Channel
+  // Join selected channel
   joinChannel = channelId => {
-    // Join Channel
+    // Join channel
     channelRepo.joinChannel({
       channelId,
-      type: EkoChannelType.Standard,
+      type: EkoChannelType.Standard
     });
     this.setState({
       currentChannelId: channelId
-    })
-  }
+    });
+  };
 
-  // Send message in Channel
+  // Send message in channel
   sendMessage = (message, channelId) => {
-    // Instantiate Message Repository
-    const messageRepo = new MessageRepository();
-    // Send Message
+    // Send message
     const messageLiveObject = messageRepo.createTextMessage({
       channelId,
-      text: message,
+      text: message
     });
     // On message sent, run the following code.
-    messageLiveObject.on('dataStatusChanged', data => {
+    messageLiveObject.on("dataStatusChanged", data => {
       console.log(`Message sent`);
     });
-  }
+  };
+
+  // Flag message
+  flagMessage = messageId => {
```

Comment:
It seems that it is time to lift all related logics down to responsible components.
flag/unflag should be located in messageItem
message create in message add component
current user (edit etc) => header or user dedicated component
flag/uflag user can leave here i think - soon we add MemberList and we will see how to structure it further.